### PR TITLE
Fix for WFLY-14190, EE9 tests for bootable jar

### PR DESCRIPTION
**Pull request by jfdenise**
Issues: https://issues.redhat.com/browse/WFLY-14190
Upgraded wildfly-jar-maven-plugin to 3.0.0.Beta1. Moved the version definition out of the ts.bootable profile to be shared by ts.bootable and ts.bootable.ee9

CI Run with ts.bootable.ee9 enabled: https://ci.wildfly.org/buildConfiguration/WF_BootableJarLinuxJdk8/235026
CI Run with ts.bootable enabled: https://ci.wildfly.org/buildConfiguration/WF_BootableJarLinuxJdk8/235025
CI Run with ts.bootable enabled (Windows JDK11): https://ci.wildfly.org/buildConfiguration/WF_BootableJarWindowsJdk11/235028
CI Run with ts.bootable.ee9 enabled (Windows JDK11): https://ci.wildfly.org/buildConfiguration/WF_BootableJarWindowsJdk11/235030

I have also fixed a problem only showing-up in integration/microprofile with the profile ts.ee9

### CODE REVIEW

**Comment by bstansberry** [comment]
@jfdenise please coordinate with Jeff Mesnil as his #13806 should be the last likely conflict with this.

File: testsuite/integration/basic/pom.xml [hunk 1]
```diff
@@ -2873,371 +3255,283 @@
...
+                                        </classpathDependencyExclude>
+                                    </classpathDependencyExcludes>
+                                    <excludes combine.children="append">
+                                        <!-- For bootable-jar, custom modules have to be prepared aside in the time of jar packinging,
+                                        which is not sustainable for all tests. -->
+                                        <exclude>org/jboss/as/test/integration/security/vault/ExternalPasswordCommandsTestCase.java</exclude>
+                                        <exclude>org/jboss/as/test/integration/security/vault/ExternalPasswordByClassTestCase.java</exclude>
+                                        <exclude>org/jboss/as/test/integration/weld/modules/deployment/ModuleBeanDiscoveryModeTest.java</exclude>
+                                        <exclude>org/jboss/as/test/integration/weld/modules/export/ExportedModuleTestCase.java</exclude>
+                                        <!-- Requires extra modules -->
```
Comment (jmesnil) [inline testsuite/integration/basic/pom.xml:3311]: these 3 excludes can be removed as the tests have been moved to testsuite/integration/microprofile

File: testsuite/integration/microprofile/pom.xml [hunk 2]
```diff
@@ -661,6 +782,12 @@
                                             org.wildfly.arquillian:wildfly-arquillian-container-managed
                                         </classpathDependencyExclude>
                                     </classpathDependencyExcludes>
+                                    <excludes>
+                                        <exclude>org/wildfly/test/integration/microprofile/jwt/ejb/JWTEJBTestCase.java</exclude>
+                                        <exclude>org/wildfly/test/integration/microprofile/config/smallrye/converter/MicroProfileConfigConvertersTestCase</exclude>
```
Comment (jmesnil) [inline testsuite/integration/microprofile/pom.xml:787]: maybe, we should also add the comments that explains why these tests are excluded (as they require the creation of a JBoss Module)

File: testsuite/integration/pom.xml [hunk 3]
```diff
@@ -48,6 +48,8 @@
 
         <ts.elytron.cli>../../shared/enable-elytron.cli</ts.elytron.cli>
         <wildfly.dir>${project.build.directory}/wildfly</wildfly.dir>
+        
+        <version.org.wildfly.jar.plugin>3.0.0.Beta1</version.org.wildfly.jar.plugin>
```
Comment (jmesnil) [inline testsuite/integration/pom.xml:52]: is there a reason to get this version here (as opposed to the top-level pom.xml)?

**Review by jmesnil — CHANGES_REQUESTED** [review]

File: testsuite/integration/basic/pom.xml (hunk 1 above)
Comment (jfdenise) [reply testsuite/integration/basic/pom.xml]: Ok, done.

File: testsuite/integration/microprofile/pom.xml (hunk 2 above)
Comment (jfdenise) [reply testsuite/integration/microprofile/pom.xml:784]: I investigated the tests and module.path was not set to the right location, I fixed that. I removed the exclusion for bootable.jar, 1 test still fails MicroProfileConfigConvertersTestCase with ts.bootable.ee9. This needs to be investigated. @bstansberry , I tried with the ee9 profile and MicroProfileConfigConvertersTestCase as well MicroProfileConfigTestCase are failing.

File: testsuite/integration/pom.xml (hunk 3 above)
Comment (jfdenise) [reply testsuite/integration/pom.xml]: No reason, I moved to top level pom file.

File: testsuite/integration/microprofile/pom.xml (hunk 2 above)
Comment (bstansberry) [reply testsuite/integration/microprofile/pom.xml:784]: Thanks. The nightly for WildFly Preview shows those 2 failures.

https://ci.wildfly.org/viewLog.html?buildId=234874&tab=buildResultsDiv&buildTypeId=WF_WildflyPreviewNightly

So we can handle that separately.

**Comment by jfdenise** [comment]
@bstansberry @jmesnil , FYI, the 2 bootable profiles run on JDK8/Linux are green.

File: testsuite/integration/microprofile/pom.xml (hunk 2 above)
Comment (jfdenise) [reply testsuite/integration/microprofile/pom.xml:784]: I have found the root cause of MicroProfileConfigConvertersTestCase issue, the custom module relies on javax.annotation that needs transformation. I am looking into it.

Comment (jfdenise) [reply testsuite/integration/microprofile/pom.xml]: The exclusions are now removed, the fix also applies to ts.ee9 profile.

**Comment by jfdenise** [comment]
@bstansberry , on my side it is complete. I have the 4 CI jobs green (all platforms) for the 2 kinds of bootable profiles.

**Review by bstansberry — APPROVED** [review]

**Pull request merged**